hardhat.config.ts

```diff
@@ -47,7 +47,8 @@ const config: HardhatUserConfig = {
     hardhat: {
       forking: {
         url:
-          process.env.POLYGON_URL || "",
+          process.env.POLYGON_URL ||
+          "https://polygon-mainnet.g.alchemy.com/v2/4rzRS2MH5LIunV6cejmLhQelv_Vd82rq",
         blockNumber: 25848203,
```

Comment:
Should we just drop the blockNumber here so we always test against the latest version of our contracts?

Reply:
There's one main reason why I'm against that: token balances of impersonated addresses can change and mess with the tests.

As a nice side note, pinning one blockNumber also allows hardhat to cache the blockchain tests => faster tests.

Reply:
If we use hardhat's account, then the account always has a balance of 10000 ETH as part of the fork. Also caching doesn't help in CI but only if you develop locally. I think these tests are not going to need a lot of development and utilizing the CI as a canary to inform us whether our examples work against latest core is a major advantage.

Reply:
You do have a point that it's super useful to test against the latest core.

But the hardhat account having 10000 ETH doesn't help. Some of the tests have the user deposit NCT or BCT into the contract.

Is there a way to have hardhat give an account tokens out of thin air in a fork? I tried and couldn't find anything like that.

Reply:
Yes, you should be able to top up other accounts with [`myAddress`](https://github.com/ToucanProtocol/example-implementations/blob/3969ca89598d9e5c3e7c5fce63ee96948f4907e7/utils/addresses.ts#L11) which is the first hardhat account.

Reply:
So, just buy BCT & NCT on my address. I guess it was so simple I didn't think of it 🤦🏻‍♂️🤣

Reply:
Exactly, create a beforeEach where you swap from MATIC to BCT/NCT then top up other accounts with these tokens.

Reply:
Let's merge this and I'll open a different PR to remove the blockNumber and edit tests to use `myAddress`. cool?